app/src/main/java/fr/neamar/kiss/ApplicationLifecycleHandler.java

```diff
@@ -0,0 +1,78 @@
+package fr.neamar.kiss;
+
+import android.app.Activity;
+import android.app.Application;
+import android.content.ComponentCallbacks2;
+import android.content.ComponentName;
+import android.content.res.Configuration;
+import android.os.Bundle;
+
+import fr.neamar.kiss.cache.MemoryCacheHelper;
+import fr.neamar.kiss.pojo.AppPojo;
+import fr.neamar.kiss.pojo.Pojo;
+
+public class ApplicationLifecycleHandler implements Application.ActivityLifecycleCallbacks, ComponentCallbacks2 {
+
+    //private static final String TAG = ApplicationLifecycleHandler.class.getSimpleName();
+    private static boolean isInBackground = false;
+
+    @Override
+    public void onActivityCreated(Activity activity, Bundle bundle) {
+    }
+
+    @Override
+    public void onActivityStarted(Activity activity) {
+    }
+
+    @Override
+    public void onActivityResumed(Activity activity) {
+
+        if (isInBackground) {
```

Comment:
Here there are two possibilities:
1. The icons aren't loaded in memory.
2. The icons are loaded in memory.

None of those involve checking if the app comes from background. 
It seems that:
```MemoryCacheHelper.cacheAppIconDrawable(activity, componentName, appPojo.userHandle);```
already checks if the icons are loaded. Maybe change the function name to ```loadAppIconDrawableIfNecessary()``` or check for a boolean (and remove the background checks).